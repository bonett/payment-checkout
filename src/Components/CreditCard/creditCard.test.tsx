import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import CreditCardComponent from "./";
import store from "Store";

describe("CreditCard component", () => {
  test("Should renders CreditCard component", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CreditCardComponent/>
        </BrowserRouter>
      </Provider>
    );

    const number = screen.getByText(/•••• •••• •••• ••••/i);
    const name = screen.getByText(/YOUR NAME HERE/i);
    expect(number).toBeInTheDocument();
    expect(name).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
