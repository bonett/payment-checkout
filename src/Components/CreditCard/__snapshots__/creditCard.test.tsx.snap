// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard component Should renders CreditCard component 1`] = `
<div>
  <div
    class="rccs"
  >
    <div
      class="rccs__card rccs__card--unknown"
    >
      <div
        class="rccs__card--front"
      >
        <div
          class="rccs__card__background"
        />
        <div
          class="rccs__issuer"
        />
        <div
          class="rccs__cvc__front"
        />
        <div
          class="rccs__number"
        >
          •••• •••• •••• ••••
        </div>
        <div
          class="rccs__name"
        >
          YOUR NAME HERE
        </div>
        <div
          class="rccs__expiry"
        >
          <div
            class="rccs__expiry__valid"
          >
            valid thru
          </div>
          <div
            class="rccs__expiry__value"
          >
            ••/••
          </div>
        </div>
        <div
          class="rccs__chip"
        />
      </div>
      <div
        class="rccs__card--back"
      >
        <div
          class="rccs__card__background"
        />
        <div
          class="rccs__stripe"
        />
        <div
          class="rccs__signature"
        />
        <div
          class="rccs__cvc"
        />
        <div
          class="rccs__issuer"
        />
      </div>
    </div>
  </div>
</div>
`;
